Cst 122 implement python s region parser in imviz

bafio opened the pull request:
Cst 122 implement python s region parser in imviz

bafio committed:
first attempt to implement a conversion of an STC-S string to a SkyRegion

bafio committed:
added ability to import STC-S region from Footprint plugin within import_region method

bmorris3 commented on jdaviz/core/region_translators.py:
```diff
@@ -353,6 +356,214 @@ def aperture2regions(aperture):
+    sky_coordinates = SkyCoord(ra, dec, frame=frame, unit=unit)
+    sky_region = PolygonSkyRegion(sky_coordinates)
+
+    # Need these for zooming
+    length = sky_coordinates[0].separation(sky_coordinates[1])
+    width = sky_coordinates[1].separation(sky_coordinates[2])
+    sky_region.height = Angle(max(length, width), unit)
+    sky_region.center = SkyCoord(ra.mean(), dec.mean(), unit=unit)
```
Let's keep these specific lines in `jwst_header_to_skyregion` in `slit_overlay.py`, since they're not always relevant for general polygons.

bmorris3 reviewed (commented):
This is awesome 💯. You did a great job following conventions in the repo. Thorough docstrings are very appreciated. Using regex here gives us space to expand/adapt this further as needed.

Only thing left to do is add a test, I'll talk with you on slack about tests.

duytnguyendtn committed:
Virtual Observatory plugin for Jdaviz (#2872)

* Migrate VO plugin to Jdaviz

* Print resource url in empty error

* Clarify degree unit on coord tooltip

* Catch connection error

* Allow URL only override

* Automatically fallback table to URL-only if unable to get metadata columns

* Coordinate frame option

* Autocenter coords on viewer if data present

* Filter resource querying on coverage at source, adjust userflow to select source before resource, add manual coordinate toggle to viewer dropdown

* Coverage filtering toggle

* Migrate resource_selected to v-model

* Mark Source and Resource as required fields

* Remove redundant manual traitlet sync

* Set throwaway event args to optional

* Drop vestigial argument signatures

* Set query registry to observe waveband_selected to avoid multiple registry calls while typing the source name

* Add TODO comment

* Black Formatting

* First draft documentation

* Fix doc references

* Fix IVOA hyperlink

* Docs: Add mention to source name string input

* Local autocenter test

* Coverage toggle test

* Move table population to testable method for local test

* Test Formatting

* Add custom table header test

* Switch to direct URL loading

* Add M51 load data test

* Add SIA2 warning

* Don't use cache to avoid loadtime warning

* Filter known data product warnings in load_data test

* Document unit tests

* Codestyle

* Changelog

* Properly compare autocenter coord values as floats

* URL fallback needs to reset current entry to URL only if previous attr have already been stored

* Table header URL fallback test

* Fix CSS height arg

* Drop vestigial ICRS frame requirement in tooltip

* Codestyle

* Syntax fix

* Only lookup source coordinate in registry query IF coverage filtering is enabled

* Define center coordinates as attr for reference in other tests

* Only block registry querying if no source and filtering enabled. Allow no source querying when filtering disabled

* Clear source field if autocoord viewer has no data

* Test comments

* Test for source error snackbar messages

* Disable registry coverage filtering by default (per Cami request)

* Test load data error messages

* Warn users about potentially misaligned data if WCS linking isn't enabled

* Load data errors do not prevent subsequent data products from loading

TEST: Combine load_data error test to main load_data test

* Test organization, codestyle

* Test that loading spinners turn off when operations are complete

* Effectively disable timeout

* Recalculate autocoord when linktype changes

* Allow float degrees

* Add support for radius units

* Use UnitSelectPluginComponent for radius unit

* Add viewer force recenter button

* Allow plugin to follow viewer coords thru panning

* Don't bother subscribing if we're on Manual

* Update test method call after renaming

* UI: Change autocenterbtn icon to indicate when coords are centered or not

* Replace Orientation Label with direct call to avoid circular import

* Change link_type to align_by

* Black Formatting

* Return on missing attr

* Replace waveband with SelectPluginComponent

* Codestyle

Co-authored-by: Kyle Conroy <kyleconroy@gmail.com>

* Ensure viewer_selected is set before centering on data

* Codestyle

Co-authored-by: Kyle Conroy <kyleconroy@gmail.com>

* Pull center value directly rather than calculate from edges

Co-authored-by: Kyle Conroy <kyleconroy@gmail.com>

* Switch manual spinner logic with decorator

* Pin pyvo to 1.5.3 bugfix

* Add reminder to remove pyvo pin upon astroquery dep update

* Add tooltip to manual resource search btn

* autocenterbtn tooltip

* Standardize Action Buttons

* Remove Common Options header

* Switch Filter Coverage button to switch

* Use zoom_center coords directly when aligned by WCS

* Reduce width of filter coverage tooltip

* Codestyle

* Implement basic plugin API

* Implement registry query api call

* Move center_on_data event check to vue method

* Remove deprecated event arg on center_on_data

* Comments

* Add missing error snackbar msg

* Add note to clarify viewer subscription regardless of tracking toggle

* Update default blank traitlet values

* Catch imposed FORMAT violation

* Set default values for required fields

* Use lambd callback and flag to determine user pan event rather than event type

* Add wide tolerances for wcs centering bug test case

* Add plugin description

* codestyle

---------

Co-authored-by: Kyle Conroy <kyleconroy@gmail.com>

kecnry committed:
deconfigged viz.loaders framework (#3439)

original framework for viz.loaders (currently hidden behind dev flag until all functionality can be built out)

kecnry committed:
loaders: data_label (#3465)

* fix AutoTextField init to set value based on default
* importers: use AutoTextField for data label
* automatically update suggested default data-label
* importers: use shared vue file where possible
* test coverage

bafio committed:
revert/adapted changes to slit_overlay.jwst_header_to_skyregion since not always relevant for general polygons
